svm/src/transaction_processor.rs

```diff
@@ -416,10 +424,9 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
                     // Load, verify and compile one program.
                     let program = load_program_with_pubkey(
                         callback,
-                        &program_cache,
+                        &self.get_environments_for_epoch(self.epoch),
```

Comment:
This introduces a nested read lock (because `program_cache` is already locked above) which is can cause a dead lock.

Reply:
Yes, created this PR to address it: https://github.com/anza-xyz/agave/pull/1616